src/Agent/NewRelic/Agent/Core/Attributes/AttributeDefinitionService.cs

```diff
@@ -476,6 +483,12 @@ public AttributeDefinitions(IAttributeFilter attribFilter)
                 .AppliesTo(AttributeDestinations.SpanEvent)
                 .Build(_attribFilter));
 
+        private AttributeDefinition<long, long> _threadId;
+        public AttributeDefinition<long, long> ThreadId => _threadId ?? (_threadId =
+            AttributeDefinitionBuilder.CreateLong("thread.id", AttributeClassification.Intrinsics)
```

Comment:
Is this really an Intrinsic attribute, since it will not always be added?

Reply:
Perhaps not, but that's how it's defined in the spec. We have other Intrinsic attributes that are conditional, so I don't think it will be a problem.